### DISCUSSION TITLE
DSCEngine functions reverting when tried interacting with it on remix..

### DISCUSSION THREAD

**Discussion by Devshenoyprasad** [Cyfrin/foundry-full-course-cu#219] [Q&A] [answered]
I tried running the Smart contract on Remix IDE.. I deployed it(sepolia). All the getter functions and view functions seems to run well. However when I try calling functions like deposit collateral it always reverts. The transferFrom is the reason the transaction is reverting. If I want to interact with the contract on Remix how can I approve the  contract to use my tokens??

**Comment by ghost** [comment, marked as answer]
You can do two things:
1. You can interact with the `WETH` token contract directly on [Sepolia Etherscan](https://sepolia.etherscan.io/address/0xdd13E55209Fd76AfE204dBda4007C227904f0a81#code) to approve `DSCEngine` as spender. 
2. Or, paste the entire `WETH` contract in a new file in Remix and use the `AtAddress` field with the address of `WETH` to load an instance of the contract, then you can interact with it.

Just in case... remember if you deployed the contract with the actual addresses of the `WETH` and `WBTC` for Sepolia then you need to mint some of those tokens by depositing `ETH`.
